src/com/owncloud/android/authentication/AuthenticatorActivity.java

```diff
@@ -878,18 +912,16 @@ private void checkBasicAuthorization() {
     }
 
     private void accessRootFolderRemoteOperation(String username, String password) {
-        Intent existenceCheckIntent = new Intent();
-        existenceCheckIntent.setAction(OperationsService.ACTION_EXISTENCE_CHECK);
-        existenceCheckIntent.putExtra(OperationsService.EXTRA_SERVER_URL, mServerInfo.mBaseUrl);
-        existenceCheckIntent.putExtra(OperationsService.EXTRA_REMOTE_PATH, "/");
-        existenceCheckIntent.putExtra(OperationsService.EXTRA_USERNAME, username);
-        existenceCheckIntent.putExtra(OperationsService.EXTRA_PASSWORD, password);
-        existenceCheckIntent.putExtra(OperationsService.EXTRA_AUTH_TOKEN, mAuthToken);
-        
-        if (mOperationsServiceBinder != null) {
-            //Log_OC.wtf(TAG, "starting existenceCheckRemoteOperation..." );
-            mWaitingForOpId = mOperationsServiceBinder.queueNewOperation(existenceCheckIntent);
+        // delete the account if the token has changed
+        if (mAction == ACTION_UPDATE_TOKEN || mAction == ACTION_UPDATE_EXPIRED_TOKEN) {

```

Comment:
Not sure this is the right place. For ACTION_UPDATE_EXPIRED_TOKEN is fine, but for ACTION_UPDATE_TOKEN... we are removing the former session before having a new valid one; what if it fails?

Need more discussion,